README.md

```diff
@@ -142,39 +99,110 @@ execute one of the following make commands. This step is not necessary in case t
+
+```c
+#include <stdio.h>
+#include <stdint.h>
+
+#include <primitives.h>
+#include <threadtools.h>
+#include <barrier.h>
+
+#define N_THREADS 10
+#define RUNS      1000000
+
+volatile int64_t object CACHE_ALIGN;
+int64_t d1 CACHE_ALIGN, d2;
+Barrier bar CACHE_ALIGN;
+
+inline static void *Execute(void *Arg) {
+    long i, id = (long)Arg;
+
+    BarrierWait(&bar);
+    if (id == 0) d1 = getTimeMillis();
+
+    for (i = 0; i < RUNS; i++)
+        FAA64(&object, 1);
+
+    BarrierWait(&bar);
+    if (id == 0) d2 = getTimeMillis();
+
+    return NULL;
+}
+
+int main(int argc, char *argv[]) {
+    object = 1;
+
+    BarrierSet(&bar, N_THREADS);
+    StartThreadsN(N_THREADS, Execute, _DONT_USE_UTHREADS_);
+    JoinThreadsN(N_THREADS - 1);
+
+    printf("time: %ld (ms)\tthroughput: %.2f (millions ops/sec)\n", 
+           (d2 - d1), RUNS * N_THREADS / (1000.0 * (d2 - d1)));
+
+    return 0;
+}
+```
+
+This example-benchmark creates `N_THREADS`, where each of them executes `RUNS` Fetch&Add operations in a shared 64-bit integer. At the end of the benchmark the throughput (i.e. Fetch&Add operations per second) is calculated.
+
+The `StartThreadsN` function (provided by the API defined in `threadtools.h`) in main, creates `N_THREADS` threads and each of the executes the `Execute` function declared in the same file. The `_DONT_USE_UTHREADS_` argument imposes `StartThreadsN` to create only Posix threads; in case that the user sets this argument to some `M` > 0, then `StartThreadsN` will create `N_THREADS` Posix threads and each of them will create `M` user-level (i.e. fiber) threads. The `JoinThreadsN` function (also provided by `threadtools. h`) waits until all Posix and fiber (if any) threads finish the execution of the `Execute` function. The Fetch&Add instruction on 64-bit integers is performed by the `FAA64` function provided by the API of `primitives.h`.
+
+The threads executing the `Execute` function use the `Barrier` re-entrant barrier object for simultaneously starting to perform Fetch&Add instructions on the shared variable `object`. This barrier is also re-used before the end of the `Execute` function in order to allow thread with `id = 0` to measure the amount of time that the benchmark needed for completion. The `BarrierSet` function in `main` initializes the `Barrier` object. The `BarrierSet` takes as an argument a pointer to the object and the number of threads `N_THREADS` that are going to use it. Both `BarrierSet` and `BarrierWait` are provided by the API of `barrier.h`
+
+At the end of the benchmark, `main` calculates and prints the average throughput of Fetch&Add operations per second achieved by the benchmark.
+
+# Performance/Optimizations
+
+Getting the best performance from the provided benchmarks is not always an easy task. Below, it follows a short-list of steps that a user could follow in order to get better performance in a modern multiprocessor.
```

Comment:
Maybe: `Below, it follows a short-list of steps that a user` --> `Here is a guide that a user`